README.md
# flutter_weather_forecast

> A weather forecast widget for Turkey.Weather data provided by the General Directorate of Meteorology.


## Table of contents
  
- [Demo](#demo) 
- [Installation](#installation)
- [Using](#using)
- [WeatherForecast](#weatherforecast)
- [License](#license)

## Demo

<img src="https://user-images.githubusercontent.com/29631083/168078939-56c3ea7f-f7e5-49f5-953d-5b03b13607be.jpg" width="300"/></a>

## Installation

```
dependencies:
  flutter_weather_forecast: ^1.1.2
```

## Using
```dart
import 'package:flutter/material.dart';
import 'package:flutter_weather_forecast/flutter_weather_forecast.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Weather Forecast by @coskuncay'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: SingleChildScrollView(
        child: Column(
          children: const [
            WeatherForecast(),
            WeatherForecast(
              stationNo: '32',
            ),
            WeatherForecast(
              stationNo: '34',
            ),
            WeatherForecast(
              stationNo: '36',
            ),
            WeatherForecast(
              stationNo: '01',
            ),
            WeatherForecast(
              stationNo: '43',
            ),
          ],
        ),
      ),
    );
  }
}
```


## WeatherForecast

| Name | Type | Default | 
| --- | --- | --- |
| stationNo | String | *required* |
| keyFontSize | double | 13 |
| valueFontSize | double | 11 |
| statusFontSize | double | 16 |
| weatherFontSize | double | 26 |
| timeFontSize | double | 13 |
| refreshIconSize | double | 20 |
| imageScale | double | .8 |
| containerHeight | double | ScreenHeight * .15 |
| borderColor | Color | Color(0XFFACAEBA) |
| keyColor | Color | Color(0XFFACAEBA) |
| valueColor | Color | Colors.black |
| weatherColor | Color | Colors.black |
| timeColor | Color | Color(0XFFACAEBA) |
| backgroundColor | Color | Colors.white |
| statusColor | Color | Colors.black |
| verticalPadding | double | 8 |
| horizontalPadding | double | 8 |

## License

[MIT](https://choosealicense.com/licenses/mit/) Copyright (c) 2022, [Emre Coşkunçay](https://github.com/coskuncay)

CHANGELOG.md
## [1.0.0] - 2022-05-12

* Initial version.

## [1.1.0] - 2022-05-12

* Custom properties added.

## [1.1.1] - 2022-05-13

* Documentation edited.

## [1.1.2] - 2022-05-25

* Container height property added.

## [1.2.0] - 2022-09-23

* Background image , boxFit , colorFilter added. DateTime padLeft added.

test/flutter_weather_forecast_test.dart
import 'package:flutter_test/flutter_test.dart';

import 'package:flutter_weather_forecast/flutter_weather_forecast.dart';

void main() {
  test('adds one to input values', () {});
}


lib/lang/tr_TR.dart
const Map<String, String> weatherStatusDict = {
  'a': 'Açık',
  'sck': 'Sıcak',
  'sgk': 'Soğuk',
  'ab': 'Az Bulutlu',
  'pb': 'Parçalı Bulutlu',
  'cb': 'Çok Bulutlu',
  'dmn': 'Duman',
  'pus': 'Pus',
  'sis ': 'Sis',
  'hy ': 'Hafif Yağmurlu',
  'y ': 'Yağmurlu',
  'ky ': 'Kuvvetli Yağmurlu',
  'hsy': 'Hafif Sağanak Yağışlı',
  'sy': 'Sağanak Yağışlı',
  'ksy': 'Kuvvetli Sağanak Yağışlı',
  'hky': 'Hafif Kar Yağışlı ',
  'k': 'Kar Yağışlı ',
  'yky': 'Yoğun Kar Yağışlı',
  'msy': 'Yer Yer Sağanak Yağışlı ',
  'dy': 'Dolu ',
  'gsy': 'Gökgürültülü Sağanak Yağışlı',
  'kky': 'Karla Karışık Yağmurlu',
  'r': 'Rüzgarlı',
  'kf': 'Toz veya Kum Fırtınası',
  'gkr': 'Güneyli Kuvvetli Rüzgar',
  'kkr': 'Kuzeyli Kuvvetli Rüzgar',
  'kgy': 'Kuvvetli Gökgürültülü Sağanak Yağışlı',
  'hhy': 'Yağış'
};


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:flutter_weather_forecast/flutter_weather_forecast.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({Key? key}) : super(key: key);
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      debugShowCheckedModeBanner: false,
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: const MyHomePage(title: 'Weather Forecast by @coskuncay'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({Key? key, required this.title}) : super(key: key);

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text(widget.title),
      ),
      body: SingleChildScrollView(
        child: Container(
          padding: const EdgeInsets.symmetric(vertical: 10, horizontal: 10),
          child: Column(
            children: [
              WeatherForecast(
                stationNo: '06',
                image: const ExactAssetImage(
                  'assets/images/ankara.jpg',
                ),
                colorFilter:
                    const ColorFilter.mode(Colors.grey, BlendMode.modulate),
                keyColor: Colors.white,
                timeColor: Colors.white,
                valueColor: Colors.white,
                statusColor: Colors.white,
                weatherColor: Colors.white,
                refreshColor: Colors.white,
              ),
              WeatherForecast(
                stationNo: '32',
              ),
              WeatherForecast(
                stationNo: '34',
                image: const ExactAssetImage(
                  'assets/images/weather_background.png',
                ),
                boxFit: BoxFit.cover,
                keyColor: Colors.white,
                timeColor: Colors.white,
                valueColor: Colors.white,
                statusColor: Colors.white,
                weatherColor: Colors.white,
                refreshColor: Colors.white,
              ),
              WeatherForecast(
                stationNo: '76',
              ),
              WeatherForecast(
                stationNo: '11',
              ),
              WeatherForecast(
                stationNo: '43',
              ),
            ],
          ),
        ),
      ),
    );
  }
}


lib/models/status_model.dart
import 'dart:convert';

class StatusModel {
  double aktuelBasinc;
  int denizSicaklik;
  double denizeIndirgenmisBasinc;
  int gorus;
  String hadiseKodu;
  int istNo;
  int kapalilik;
  int karYukseklik;
  int nem;
  String rasatMetar;
  String rasatSinoptik;
  String rasatTaf;
  double ruzgarHiz;
  int ruzgarYon;
  double sicaklik;
  String veriZamani;
  double yagis00Now;
  int yagis10Dk;
  double yagis12Saat;
  int yagis1Saat;
  double yagis24Saat;
  double yagis6Saat;
  String denizVeriZamani;
  StatusModel({
    required this.aktuelBasinc,
    required this.denizSicaklik,
    required this.denizeIndirgenmisBasinc,
    required this.gorus,
    required this.hadiseKodu,
    required this.istNo,
    required this.kapalilik,
    required this.karYukseklik,
    required this.nem,
    required this.rasatMetar,
    required this.rasatSinoptik,
    required this.rasatTaf,
    required this.ruzgarHiz,
    required this.ruzgarYon,
    required this.sicaklik,
    required this.veriZamani,
    required this.yagis00Now,
    required this.yagis10Dk,
    required this.yagis12Saat,
    required this.yagis1Saat,
    required this.yagis24Saat,
    required this.yagis6Saat,
    required this.denizVeriZamani,
  });

  static StatusModel fromJsonModel(Map<String, dynamic> map) =>
      StatusModel.fromMap(map);

  Map<String, dynamic> toMap() {
    return {
      'aktuelBasinc': aktuelBasinc,
      'denizSicaklik': denizSicaklik,
      'denizeIndirgenmisBasinc': denizeIndirgenmisBasinc,
      'gorus': gorus,
      'hadiseKodu': hadiseKodu,
      'istNo': istNo,
      'kapalilik': kapalilik,
      'karYukseklik': karYukseklik,
      'nem': nem,
      'rasatMetar': rasatMetar,
      'rasatSinoptik': rasatSinoptik,
      'rasatTaf': rasatTaf,
      'ruzgarHiz': ruzgarHiz,
      'ruzgarYon': ruzgarYon,
      'sicaklik': sicaklik,
      'veriZamani': veriZamani,
      'yagis00Now': yagis00Now,
      'yagis10Dk': yagis10Dk,
      'yagis12Saat': yagis12Saat,
      'yagis1Saat': yagis1Saat,
      'yagis24Saat': yagis24Saat,
      'yagis6Saat': yagis6Saat,
      'denizVeriZamani': denizVeriZamani,
    };
  }

  factory StatusModel.fromMap(Map<String, dynamic> map) {
    return StatusModel(
      aktuelBasinc: map['aktuelBasinc']?.toDouble() ?? 0.0,
      denizSicaklik: map['denizSicaklik']?.toInt() ?? 0,
      denizeIndirgenmisBasinc:
          map['denizeIndirgenmisBasinc']?.toDouble() ?? 0.0,
      gorus: map['gorus']?.toInt() ?? 0,
      hadiseKodu: map['hadiseKodu'] ?? '',
      istNo: map['istNo']?.toInt() ?? 0,
      kapalilik: map['kapalilik']?.toInt() ?? 0,
      karYukseklik: map['karYukseklik']?.toInt() ?? 0,
      nem: map['nem']?.toInt() ?? 0,
      rasatMetar: map['rasatMetar'] ?? '',
      rasatSinoptik: map['rasatSinoptik'] ?? '',
      rasatTaf: map['rasatTaf'] ?? '',
      ruzgarHiz: map['ruzgarHiz']?.toDouble() ?? 0.0,
      ruzgarYon: map['ruzgarYon']?.toInt() ?? 0,
      sicaklik: map['sicaklik']?.toDouble() ?? 0.0,
      veriZamani: map['veriZamani'] ?? '',
      yagis00Now: map['yagis00Now']?.toDouble() ?? 0.0,
      yagis10Dk: map['yagis10Dk']?.toInt() ?? 0,
      yagis12Saat: map['yagis12Saat']?.toDouble() ?? 0.0,
      yagis1Saat: map['yagis1Saat']?.toInt() ?? 0,
      yagis24Saat: map['yagis24Saat']?.toDouble() ?? 0.0,
      yagis6Saat: map['yagis6Saat']?.toDouble() ?? 0.0,
      denizVeriZamani: map['denizVeriZamani'] ?? '',
    );
  }

  String toJson() => json.encode(toMap());

  factory StatusModel.fromJson(String source) =>
      StatusModel.fromMap(json.decode(source));
}


lib/models/station_model.dart
import 'dart:convert';

class StationModel {
  dynamic alternatifHadiseIstNo;

  /// longitude of station
  double boylam;

  /// latitude of station
  double enlem;

  /// station code
  int gunlukTahminIstNo;

  /// city
  String il;

  /// city code
  int ilPlaka;

  /// county
  String ilce;

  ///center county code
  int merkezId;
  int oncelik;
  int saatlikTahminIstNo;
  int sondurumIstNo;

  ///altitude
  int yukseklik;

  /// description
  String aciklama;
  int modelId;
  int gps;
  StationModel({
    required this.alternatifHadiseIstNo,
    required this.boylam,
    required this.enlem,
    required this.gunlukTahminIstNo,
    required this.il,
    required this.ilPlaka,
    required this.ilce,
    required this.merkezId,
    required this.oncelik,
    required this.saatlikTahminIstNo,
    required this.sondurumIstNo,
    required this.yukseklik,
    required this.aciklama,
    required this.modelId,
    required this.gps,
  });

  Map<String, dynamic> toMap() {
    return {
      'alternatifHadiseIstNo': alternatifHadiseIstNo,
      'boylam': boylam,
      'enlem': enlem,
      'gunlukTahminIstNo': gunlukTahminIstNo,
      'il': il,
      'ilPlaka': ilPlaka,
      'ilce': ilce,
      'merkezId': merkezId,
      'oncelik': oncelik,
      'saatlikTahminIstNo': saatlikTahminIstNo,
      'sondurumIstNo': sondurumIstNo,
      'yukseklik': yukseklik,
      'aciklama': aciklama,
      'modelId': modelId,
      'gps': gps,
    };
  }

  factory StationModel.fromMap(Map<String, dynamic> map) {
    return StationModel(
      alternatifHadiseIstNo: map['alternatifHadiseIstNo'] ?? null,
      boylam: map['boylam']?.toDouble() ?? 0.0,
      enlem: map['enlem']?.toDouble() ?? 0.0,
      gunlukTahminIstNo: map['gunlukTahminIstNo']?.toInt() ?? 0,
      il: map['il'] ?? '',
      ilPlaka: map['ilPlaka']?.toInt() ?? 0,
      ilce: map['ilce'] ?? '',
      merkezId: map['merkezId']?.toInt() ?? 0,
      oncelik: map['oncelik']?.toInt() ?? 0,
      saatlikTahminIstNo: map['saatlikTahminIstNo']?.toInt() ?? 0,
      sondurumIstNo: map['sondurumIstNo']?.toInt() ?? 0,
      yukseklik: map['yukseklik']?.toInt() ?? 0,
      aciklama: map['aciklama'] ?? '',
      modelId: map['modelId']?.toInt() ?? 0,
      gps: map['gps']?.toInt() ?? 0,
    );
  }

  String toJson() => json.encode(toMap());

  factory StationModel.fromJson(String source) =>
      StationModel.fromMap(json.decode(source));
}


lib/models/weather_model.dart
import 'dart:convert';

class WeatherModel {
  /// current weather forecast
  String? weatherCode;

  /// minimum temperature of today's
  String? minTemp;

  /// maximum temperature of today's
  String? maxTemp;

  /// wind speed of today's
  String? windSpeed;

  /// humidity of today's
  String? hum;

  /// current temperature of today's
  String? currentTemp;

  WeatherModel({
    this.weatherCode,
    this.minTemp,
    this.maxTemp,
    this.windSpeed,
    this.hum,
    this.currentTemp,
  });

  Map<String, dynamic> toMap() {
    return {
      'weatherCode': weatherCode,
      'minTemp': minTemp,
      'maxTemp': maxTemp,
      'windSpeed': windSpeed,
      'hum': hum,
      'currentTemp': currentTemp,
    };
  }

  factory WeatherModel.fromMap(Map<String, dynamic> map) {
    return WeatherModel(
      weatherCode: map['weatherCode'] ?? '',
      minTemp: map['minTemp'] ?? '',
      maxTemp: map['maxTemp'] ?? '',
      windSpeed: map['windSpeed'] ?? '',
      hum: map['hum'] ?? '',
      currentTemp: map['currentTemp'] ?? '',
    );
  }

  String toJson() => json.encode(toMap());

  factory WeatherModel.fromJson(String source) =>
      WeatherModel.fromMap(json.decode(source));
}


lib/flutter_weather_forecast.dart
library flutter_weather_forecast;

import 'dart:convert';

import 'package:flutter_weather_forecast/lang/tr_TR.dart';
import 'package:flutter_weather_forecast/models/forecast_model.dart';
import 'package:flutter_weather_forecast/models/status_model.dart';
import 'package:http/http.dart' as http;
import 'package:flutter/material.dart';

import 'models/weather_model.dart';

// ignore: must_be_immutable
class WeatherForecast extends StatefulWidget {
  WeatherForecast({
    Key? key,
    required this.stationNo,
    this.keyFontSize = 13.0,
    this.valueFontSize = 13.0,
    this.statusFontSize = 14.0,
    this.weatherFontSize = 26.0,
    this.timeFontSize = 13.0,
    this.refreshIconSize = 20.0,
    this.imageScale = .8,
    this.borderColor = const Color(0XFFACAEBA),
    this.keyColor = const Color(0XFFACAEBA),
    this.valueColor = Colors.black,
    this.statusColor = Colors.black,
    this.weatherColor = Colors.black,
    this.refreshColor = Colors.black,
    this.timeColor = const Color(0XFFACAEBA),
    this.backgroundColor = Colors.white,
    this.verticalPadding = 8.0,
    this.colorFilter,
    this.horizontalPadding = 8.0,
    this.containerHeight,
    this.image,
    this.boxFit = BoxFit.cover,
  }) : super(key: key);

  /// City code. Default is 06
  final String stationNo;

  /// Font size of Min,Max.. keys. Default is 13
  final double keyFontSize;

  /// Font size of temp,humidity .. values. Default is 11
  final double valueFontSize;

  /// Font size of status. Default is 16
  final double statusFontSize;

  /// Font size of status.Default is 26
  final double weatherFontSize;

  /// Font size of time.Default is 13
  final double timeFontSize;

  /// Size of Refresh Icon.Default is 20
  final double refreshIconSize;

  /// Scale of Image. Default .8
  final double imageScale;

  /// Height of container. Default ScreenHeight * .15
  final double? containerHeight;

  /// [Color] of selected dot. Default is Color(0XFFACAEBA)
  final Color borderColor;

  /// [Color] of selected dot. Default is Color(0XFFACAEBA)
  final Color keyColor;

  /// [Color] of selected dot. Default is  [Colors.black]
  final Color valueColor;

  /// [Color] of selected dot. Default is [Colors.black]
  final Color weatherColor;

  /// [Color] of selected dot. Default is [Colors.black]
  final Color refreshColor;

  /// [Color] of selected dot. Default is Color(0XFFACAEBA)
  final Color timeColor;

  /// [Color] of selected dot. Default is [Colors.white]
  final Color backgroundColor;

  /// [Color] of selected dot. Default is  [Colors.black]
  final Color statusColor;

  /// Vertical Padding . Default 8
  final double verticalPadding;

  /// Horizontal Padding . Default 8
  final double horizontalPadding;

  // ignore: prefer_typing_uninitialized_variables
  var image;

  /// [BoxFit] of image. Default is  [BoxFit.fill]
  final BoxFit boxFit;

  /// [ColorFilter] of image. Default is null
  ColorFilter? colorFilter;

  @override
  State<WeatherForecast> createState() => _WeatherForecastState();
}

class _WeatherForecastState extends State<WeatherForecast> {
  bool isLoading = false;
  bool hasError = false;
  late String errorMessage;
  late WeatherModel weatherModel;
  late String requestTime;
  @override
  void initState() {
    weatherModel = WeatherModel();
    super.initState();
    getCurrentForecast();
  }

  Future<void> getForecastDaily() async {
    final queryParameters = {
      'istno': '9' + widget.stationNo + '01',
    };
    final uri = Uri.https(
        'servis.mgm.gov.tr', '/web/tahminler/gunluk', queryParameters);
    var response =
        await http.get(uri, headers: {'origin': 'https://www.mgm.gov.tr'});
    try {
      ForecastModel dailyForecast =
          ForecastModel.fromJsonModel(json.decode(response.body)[0]);
      setState(
        () {
          weatherModel.minTemp = dailyForecast.enDusukGun1.toString();
          weatherModel.maxTemp = dailyForecast.enYuksekGun1.toString();
        },
      );
    } catch (e) {
      print(e);
    }
  }

  Future<void> getCurrentForecast() async {
    DateTime now = DateTime.now();
    setState(() {
      requestTime = now.day.toString().padLeft(2, '0') +
          "." +
          now.month.toString().padLeft(2, '0') +
          "." +
          now.year.toString() +
          "-" +
          now.hour.toString().padLeft(2, '0') +
          ":" +
          now.minute.toString().padLeft(2, '0');
    });
    final queryParameters = {
      'merkezid': '9' + widget.stationNo + '01',
    };
    final uri =
        Uri.https('servis.mgm.gov.tr', '/web/sondurumlar', queryParameters);
    var response =
        await http.get(uri, headers: {'origin': 'https://www.mgm.gov.tr'});
    try {
      StatusModel currentForecast =
          StatusModel.fromJsonModel(json.decode(response.body)[0]);
      setState(
        () {
          weatherModel.weatherCode = currentForecast.hadiseKodu;
          weatherModel.currentTemp =
              currentForecast.sicaklik.toStringAsFixed(1);
          weatherModel.windSpeed = currentForecast.ruzgarHiz.toInt().toString();
          weatherModel.hum = currentForecast.nem.toInt().toString();
          getForecastDaily();
          isLoading = false;
          hasError = false;
        },
      );
    } catch (e) {
      setState(
        () {
          hasError = true;
          isLoading = false;
          errorMessage = "Hava durumu bilgileri alınamadı! Tekrar deneyiniz.";
        },
      );
    }
  }

  @override
  Widget build(BuildContext context) {
    return Padding(
      padding: EdgeInsets.symmetric(
        vertical: widget.verticalPadding,
        horizontal: widget.horizontalPadding,
      ),
      child: Column(
        crossAxisAlignment: CrossAxisAlignment.center,
        children: [
          Container(
            height: widget.containerHeight ??
                MediaQuery.of(context).size.height * .15,
            decoration: widget.image != null
                ? BoxDecoration(
                    borderRadius: BorderRadius.circular(5),
                    image: DecorationImage(
                      image: widget.image,
                      colorFilter: widget.colorFilter,
                      fit: widget.boxFit,
                    ),
                  )
                : BoxDecoration(
                    borderRadius: BorderRadius.circular(5),
                    color: widget.backgroundColor,
                    border: Border.all(
                      color: hasError ? Colors.red : widget.borderColor,
                      width: 1,
                    ),
                  ),
            child: Padding(
              padding: const EdgeInsets.all(8.0),
              child: hasError
                  ? Center(
                      child: Column(
                        crossAxisAlignment: CrossAxisAlignment.center,
                        mainAxisAlignment: MainAxisAlignment.center,
                        children: [
                          Text(errorMessage),
                          isLoading
                              ? const SizedBox(
                                  width: 20,
                                  height: 20,
                                  child: CircularProgressIndicator(
                                    strokeWidth: 1,
                                    color: Color.fromARGB(255, 177, 177, 177),
                                  ),
                                )
                              : InkWell(
                                  onTap: () async {
                                    setState(
                                      () {
                                        isLoading = true;
                                        getCurrentForecast();
                                      },
                                    );
                                  },
                                  child: Icon(
                                    Icons.refresh,
                                    size: widget.refreshIconSize,
                                    color: widget.refreshColor,
                                  ),
                                ),
                        ],
                      ),
                    )
                  : (weatherModel.currentTemp == null ||
                          weatherModel.minTemp == null)
                      ? const Center(
                          child: CircularProgressIndicator(
                            strokeWidth: 1,
                          ),
                        )
                      : Column(
                          mainAxisAlignment: MainAxisAlignment.start,
                          crossAxisAlignment: CrossAxisAlignment.center,
                          children: [
                            Row(
                              mainAxisAlignment: MainAxisAlignment.end,
                              children: [
                                isLoading
                                    ? const SizedBox(
                                        width: 5,
                                        height: 10,
                                        child: CircularProgressIndicator(
                                          strokeWidth: 1,
                                          color: Color.fromARGB(
                                              255, 177, 177, 177),
                                        ),
                                      )
                                    : InkWell(
                                        onTap: () {
                                          setState(
                                            () {
                                              isLoading = true;
                                              getCurrentForecast();
                                            },
                                          );
                                        },
                                        child: Icon(
                                          Icons.refresh,
                                          size: widget.refreshIconSize,
                                          color: widget.refreshColor,
                                        ),
                                      ),
                                Text(
                                  requestTime,
                                  textAlign: TextAlign.right,
                                  style: TextStyle(
                                    color: widget.timeColor,
                                    fontFamily: 'Montserrat',
                                    fontSize: widget.timeFontSize,
                                    fontWeight: FontWeight.normal,
                                    fontStyle: FontStyle.italic,
                                  ),
                                ),
                              ],
                            ),
                            Row(
                              children: [
                                Image.asset(
                                  'assets/images/' +
                                      weatherModel.weatherCode!.toLowerCase() +
                                      '.png',
                                  package: 'flutter_weather_forecast',
                                  scale: widget.imageScale,
                                ),
                                const SizedBox(
                                  width: 10,
                                ),
                                Column(
                                  mainAxisAlignment: MainAxisAlignment.start,
                                  crossAxisAlignment: CrossAxisAlignment.start,
                                  children: [
                                    // Figma Flutter Generator GneliWidget - TEXT
                                    weatherStatusDict[weatherModel.weatherCode!
                                                .toLowerCase()] ==
                                            null
                                        ? Text(
                                            weatherStatusDict['ab']!,
                                            textAlign: TextAlign.left,
                                            style: TextStyle(
                                              color: widget.statusColor,
                                              fontFamily: 'Montserrat',
                                              fontSize: widget.statusFontSize,
                                              fontWeight: FontWeight.normal,
                                            ),
                                          )
                                        : SizedBox(
                                            width: MediaQuery.of(context)
                                                    .size
                                                    .width *
                                                .2,
                                            child: Text(
                                              weatherStatusDict[weatherModel
                                                  .weatherCode!
                                                  .toLowerCase()]!,
                                              textAlign: TextAlign.left,
                                              overflow: TextOverflow.ellipsis,
                                              style: TextStyle(
                                                color: widget.statusColor,
                                                fontFamily: 'Montserrat',
                                                fontSize: widget.statusFontSize,
                                                fontWeight: FontWeight.normal,
                                              ),
                                            ),
                                          ),
                                    Text(
                                      weatherModel.currentTemp! + '°C',
                                      textAlign: TextAlign.left,
                                      style: TextStyle(
                                        fontSize: widget.weatherFontSize,
                                        fontWeight: FontWeight.normal,
                                        color: widget.weatherColor,
                                      ),
                                    ),
                                  ],
                                ),
                                const SizedBox(
                                  width: 10,
                                ),
                                Padding(
                                  padding:
                                      const EdgeInsets.symmetric(vertical: 20),
                                  child: Column(
                                    crossAxisAlignment:
                                        CrossAxisAlignment.start,
                                    mainAxisAlignment: MainAxisAlignment.center,
                                    children: [
                                      Row(
                                        children: [
                                          Text(
                                            'Min: ',
                                            textAlign: TextAlign.right,
                                            style: TextStyle(
                                              color: widget.keyColor,
                                              fontSize: widget.keyFontSize,
                                              fontStyle: FontStyle.italic,
                                            ),
                                          ),
                                          Text(
                                            weatherModel.minTemp! + ' °C',
                                            style: TextStyle(
                                              color: widget.valueColor,
                                              fontFamily: 'Montserrat',
                                              fontSize: widget.valueFontSize,
                                              fontStyle: FontStyle.italic,
                                            ),
                                          ),
                                        ],
                                      ),
                                      const SizedBox(
                                        height: 10,
                                      ),
                                      Row(
                                        children: [
                                          Text(
                                            'Max: ',
                                            textAlign: TextAlign.right,
                                            style: TextStyle(
                                              color: widget.keyColor,
                                              fontFamily: 'Montserrat',
                                              fontSize: widget.keyFontSize,
                                              fontStyle: FontStyle.italic,
                                            ),
                                          ),
                                          Text(
                                            weatherModel.maxTemp! + ' °C',
                                            style: TextStyle(
                                              color: widget.valueColor,
                                              fontFamily: 'Montserrat',
                                              fontSize: 12,
                                              fontStyle: FontStyle.italic,
                                            ),
                                          ),
                                        ],
                                      ),
                                    ],
                                  ),
                                ),
                                const SizedBox(
                                  width: 10,
                                ),
                                Column(
                                  crossAxisAlignment: CrossAxisAlignment.start,
                                  mainAxisAlignment: MainAxisAlignment.center,
                                  children: [
                                    Row(
                                      children: [
                                        Text(
                                          'Rüzgar: ',
                                          textAlign: TextAlign.right,
                                          style: TextStyle(
                                            color: widget.keyColor,
                                            fontSize: widget.keyFontSize,
                                            fontStyle: FontStyle.italic,
                                          ),
                                        ),
                                        Text(
                                          weatherModel.windSpeed! + 'km/h',
                                          style: TextStyle(
                                            color: widget.valueColor,
                                            fontFamily: 'Montserrat',
                                            fontSize: widget.valueFontSize,
                                            fontStyle: FontStyle.italic,
                                          ),
                                        ),
                                      ],
                                    ),
                                    const SizedBox(
                                      height: 10,
                                    ),
                                    Row(
                                      children: [
                                        Text(
                                          'Nem: ',
                                          textAlign: TextAlign.right,
                                          style: TextStyle(
                                            color: widget.keyColor,
                                            fontFamily: 'Montserrat',
                                            fontSize: widget.keyFontSize,
                                            fontStyle: FontStyle.italic,
                                          ),
                                        ),
                                        Text(
                                          weatherModel.hum! + '%',
                                          style: TextStyle(
                                            color: widget.valueColor,
                                            fontFamily: 'Montserrat',
                                            fontSize: widget.valueFontSize,
                                            fontStyle: FontStyle.italic,
                                          ),
                                        ),
                                      ],
                                    ),
                                  ],
                                ),
                              ],
                            ),
                          ],
                        ),
            ),
          ),
        ],
      ),
    );
  }
}
